### DISCUSSION TITLE
[Feature] Edit instead of view

### DISCUSSION THREAD

**Discussion by rtritto** [TexteaInc/json-viewer#328] [Ideas]
At first load, _Edit_ the JSON instead of _View_.

e.g.:
- JSON in _edit mode_:
```js
{ <ICON_TO_SAVE_CHANGES>
  field1: 'value1',
  ...
}
```

- type and click <ICON_TO_SAVE_CHANGES> to save

**Comment by pionxzh** [comment]
Sorry. I don't quite understand what you were describing.
1. Change the position of "Save"?
2. Text-based object editor?
3. Switch to edit mode programatically/on load?

**Reply by rtritto** [reply]
The purpose is a JSON editor.
So first edit text (text is editable on load), then click on Save icon.

**Reply by pionxzh** [reply]
I still don't understand 😢 
Like all the fields will be in edit mode by default?

For this object, where is the save button?
```
{
   foo: 'bar',
   field1: 'value1',
   field2: 'value2',
}

**Reply by rtritto** [reply]
> Like all the fields will be in edit mode by default?

Yes.

> For this object, where is the save button?
> ```
> {
>    foo: 'bar',
>    field1: 'value1',
>    field2: 'value2',
> }
> ```

It's needed a button to save all fields (should be possible to add a field or change its type).
I would put it near the opened curly bracket.

**Reply by pionxzh** [reply]
For now, it can be achieved by extending all the built-in data types with your own editor. And you can inject "Save" button via providing a special data type which matched the outer object with `PreComponent`.

**Reply by rtritto** [reply]
What should I use for the `PreComponent`?

**Reply by pionxzh** [reply]
A `{` and the "<ICON_TO_SAVE_CHANGES>" button.
